miner/block_builder.go

```diff
@@ -180,9 +182,15 @@ func (t *BlockBuilder) createBlock(id types.LayerID, atxID types.AtxId, eligibil
 	} else if id == config.Genesis+1 {
 		votes = append(votes, config.GenesisId)
 	} else {
-		votes, err = t.hareResult.GetResult(id - 1)
+
+		bottom := types.LayerID(1)
+		if id > t.hdist {
+			bottom = id - t.hdist
```

Comment:
The range of layers includes both the bottom and the id (as the higher limit). In such case this line should be `bottom = id -t.hdist + 1`